src/main/java/org/radarcns/webapp/SubjectEndPoint.java

```diff
@@ -145,6 +150,7 @@ public Response getSubjectJson(
             @PathParam(SUBJECT_ID) String subject
     ) {
         try {
+            checkPermission(getJWT(request), SUBJECT_READ);
```

Comment:
Is an endpoint to get **all** subjects necessary? This should be discussed in the wider group of developers. An endpoint like this still exists in ManagementPortal as well and should be fixed there too. It will give access to read all subjects to anyone with a `SUBJECT_READ` permission, since there is no 'project' limitation. If such an endpoint is desired, we should probably add a dedicated permission for it. e.g. `SUBJECT_READ_ALL`.

Reply:
I think this endpoint is not entirely necessary. But the dashboard may require it. Maybe @herkulano can clear this. And yes should have a `SUBJECT_READ_ALL` permission if required